### PR TITLE
Support JWT authentication

### PR DESCRIPTION
**Pull request by hilott**
Support JWT Authentication

### BREAKING CHANGES
- rename TimeTreeClient to OAuthClient
- rename Authenticator to OAuthAuthenticator

### CODE REVIEW

File: web-api/package.json [hunk 1]
```diff
@@ -45,6 +46,7 @@
     "@types/async-retry": "1.4.2",
     "@types/humps": "2.0.0",
     "@types/jest": "26.0.13",
+    "@types/jsonwebtoken": "^8.5.0",
```
Comment (fujikky) [inline web-api/package.json:49]: Should this be a pinned version?

File: web-api/src/calendarApp/CalendarAppAuthenticator.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      }
+    );
+    return response.data;
+  }
+
+  private generateToken() {
+    const now = Math.floor(new Date().getTime() / 1000);
+    const payload = {
+      iat: now,
+      exp:
```
Comment (hilott) [inline web-api/src/calendarApp/CalendarAppAuthenticator.ts:65]: The exp claim identifies the expiration time.
Number of **seconds** from 1970-01-01T00:00:00Z UTC until the specified UTC date/time, ignoring leap seconds.
https://tools.ietf.org/html/rfc7519#section-4.1.4

File: web-api/package.json (hunk 1 above)
Comment (hilott) [reply web-api/package.json]: Yes, I fixed it!
https://github.com/jubilee-works/timetree-sdk-js/pull/250/commits/d6a1f49ef9bcd02a45c6c1fa82680b3dfa308e6f

**Review by fujikky — APPROVED** [review]
LGTM 👍

**Pull request merged**